src/library/scala/concurrent/Future.scala

```diff
@@ -618,7 +664,7 @@ object Future {
    */
   def sequence[A, M[X] <: TraversableOnce[X]](in: M[Future[A]])(implicit cbf: CanBuildFrom[M[Future[A]], A, M[A]], executor: ExecutionContext): Future[M[A]] = {
     in.foldLeft(successful(cbf(in))) {
-      (fr, fa) => for (r <- fr; a <- fa) yield (r += a)
+      (fr, fa) => fr.zipWith(fa)(_ += _)
```

Comment:
@retronym Speaking of this line btw, will the compiler hoist the allocation since it is a pure function? (I'm going to assume: no, in which case it perhaps ought to be manually hoisted) (I have no access to the bytecode on this machine)